src/lib/modelUtils.ts

```diff
@@ -0,0 +1,335 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { DataTypes, Model } from 'sequelize';
+import * as dotWild from 'dot-wild';
+import merge from 'deepmerge';
+
+// Define the mapping between Sequelize data types and TypeScript data types
+const dataTypeMapping = {
+  [DataTypes.ABSTRACT.key]: undefined,
+  [DataTypes.STRING.key]: 'string',
+  [DataTypes.CHAR.key]: 'string',
+  [DataTypes.TEXT.key]: 'string',
+  [DataTypes.NUMBER.key]: 'number',
+  [DataTypes.TINYINT.key]: 'number',
+  [DataTypes.SMALLINT.key]: 'number',
+  [DataTypes.MEDIUMINT.key]: 'number',
+  [DataTypes.INTEGER.key]: 'number',
+  [DataTypes.BIGINT.key]: 'number',
+  [DataTypes.FLOAT.key]: 'number',
+  [DataTypes.REAL.key]: 'number',
+  [DataTypes.DOUBLE.key]: 'number',
+  [DataTypes.DECIMAL.key]: 'number',
+  [DataTypes.BOOLEAN.key]: 'boolean',
+  [DataTypes.TIME.key]: 'string',
+  [DataTypes.DATE.key]: 'string',
+  [DataTypes.DATEONLY.key]: 'string',
+  [DataTypes.HSTORE.key]: undefined,
+  [DataTypes.JSON.key]: 'object',
+  [DataTypes.JSONB.key]: 'object',
+  [DataTypes.NOW.key]: undefined,
+  [DataTypes.BLOB.key]: undefined,
```

Comment:
So these undefined mappings simply don't have a corresponding value?

Reply:
correct

Reply:
the list is all the types defined in the sequelize.DataTypes. but many are very rarely ever used in any context, and not in our context. The only one that is marked as unknow that we use on any table we use is DataType.ARRAY. To have a mapping for that I would also need to identify the type of the item in the array. So I think this is a not going to be as problem as all the tables in this PR don't use the array datatype.